bin/copyQCDataToTmp.sh

```diff
@@ -0,0 +1,389 @@
+		# and a second time without just to make sure we can use the content from the sourced files.
+		#
+		# Disable shellcheck code syntax checking for config files.
+		# shellcheck source=/dev/null
+		mixed_stdouterr=$(source "${configFile}" 2>&1) || log4Bash 'FATAL' "${LINENO}" "${FUNCNAME[0]:-main}" "${?}" "Cannot source ${configFile}."
+		# shellcheck source=/dev/null
+		source "${configFile}"  # May seem redundant, but is a mandatory workaround for some Bash versions.
+	else
+		log4Bash 'FATAL' "${LINENO}" "${FUNCNAME[0]:-main}" '1' "Config file ${configFile} missing or not accessible."
+	fi
+done
+
+if [[ "${ROLE_USER}" != "${DATA_MANAGER}" ]]
+then
+	log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "This script must be executed by user ${DATA_MANAGER}, but you are ${ROLE_USER} (${REAL_USER})."
+fi
+
+infoServerLocation="${HOSTNAME_PRM}"
+infoLocation="/groups/${group}/${PRM_LFS}/trendanalysis/"
+hashedSource="$(printf '%s:%s' "${infoServerLocation}" "${infoLocation}" | md5sum | awk '{print $1}')"
+lockFile="${WORKING_DIR}/trendanalysis/logs/${SCRIPT_NAME}_${hashedSource}.lock"
+thereShallBeOnlyOne "${lockFile}"
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Successfully got exclusive access to lock file ${lockFile} ..."
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Log files will be written to ${PRM_ROOT_DIR}/trendanalysis/logs/ ..."
+
+
+#
+## Loops through all rawdata folders and checks if the QC data  is already copied to tmp. If not than call function copyQCRawdataToTmp
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "starting checking the prm's for raw QC data"
+
+for prm_dir in "${ALL_PRM[@]}"
+do
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "looping through ${prm_dir}"
+	export PRM_ROOT_DIR="/groups/${group}/${prm_dir}/"
+	readarray -t rawdata < <(find "${PRM_ROOT_DIR}/rawdata/ngs/" -maxdepth 1 -mindepth 1 -type d -name "[!.]*" | sed -e "s|^${PRM_ROOT_DIR}/rawdata/ngs/||")
+
+	if [[ "${#rawdata[@]}" -eq '0' ]]
+	then
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "No rawdata found @ ${PRM_ROOT_DIR}/rawdata/ngs/."
+	else
+		for rawdat in "${rawdata[@]}"
+		do
+			log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing rawdat ${rawdat} ..."
+			if  [[ -d "${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/" ]]
+			then
+				controlFileBase="${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
+				export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${rawdat}.${SCRIPT_NAME}"
+				log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing run ${rawdat} ..."
+			else
+				mkdir "${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
+				log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Creating logs folder: ${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
+				controlFileBase="${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
+				export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${rawdat}.${SCRIPT_NAME}"
+				log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing run ${rawdat} ..."
+			fi
```

Comment:
you could also do `mkdir -p "${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"` then you don't need an if else:

```
mkdir -p "${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Creating logs folder: ${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
controlFileBase="${PRM_ROOT_DIR}/trendanalysis/logs/${rawdat}/"
export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${rawdat}.${SCRIPT_NAME}"
log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Processing run ${rawdat} ..."
```